src/server/frontend_wayland/wl_surface.h

```diff
@@ -101,6 +101,20 @@ class NullWlSurfaceRole : public WlSurfaceRole
     WlSurface* const surface;
 };
 
+class DragWlSurfaceRole : public NullWlSurfaceRole
+{
+public:
+    DragWlSurfaceRole(WlSurface* surface);
+    ~DragWlSurfaceRole();
+    auto scene_surface() const -> std::optional<std::shared_ptr<scene::Surface>> override;
+    void commit(WlSurfaceState const& state) override;
+    void create_scene_surface();
+
+private:
+    wayland::Weak<WlSurface> const surface;
+    std::weak_ptr<scene::Surface> weak_scene_surface;
```

Comment:
It's not clear to me what value having this be a `weak_ptr` gives; it's added to the scene immediately, which holds a strong reference until something removes it from the scene, and the way *that* happens is either in `~DragWlSurfaceRole()` or on client destruction.

In any case, this seems like it'd be happer as a `shared_ptr`?